lib/ansible/plugins/action/set_fact.py

```diff
@@ -45,10 +44,6 @@ def run(self, tmp=None, task_vars=None):
             for (k, v) in self._task.args.items():
                 k = self._templar.template(k)
 
-                if not isidentifier(k):
-                    raise AnsibleActionFail("The variable name '%s' is not valid. Variables must start with a letter or underscore character, "
-                                            "and contain only letters, numbers and underscores." % k)
-
```

Comment:
I would still fail on 'is identifier', as this always causes issues and was already the behavior here.

Reply:
It still fails but the check is now in vars manager in `set_nonpersistent_facts` and/or `set_host_facts`.

Reply:
doesnt that depend on a toggle now?

Reply:
Only reserved keywords are behind the toggle. Not passing `isidentifier` always results in an error.

Reply:
ah, misread the code, then ignoring this is fine